import of the vmware_rest documentation (#72090)
ansible/ansible: docs/docsite/rst/scenario_guides/vmware_rest_scenarios/create_vm.rst

.. _vmware_rest_create_vm:

*******************************
How to create a Virtual Machine
*******************************

> .. contents:: Topics

Reviewer: ```suggestion
.. contents::
  :local:

```

Introduction
============

> This guide will show you how to utilize Ansible to create a virtual machine.

Reviewer: ```suggestion
This section shows you how to use Ansible to create a virtual machine.
```

> Scenario Requirements

Reviewer: ```suggestion
Scenario requirements
```

=====================

You've already followed :ref:`vmware_rest_collect_info` and you've got the following variables defined:

- ``my_cluster_info``
- ``my_datastore``
- ``my_virtual_machine_folder``
- ``my_cluster_info``

> How to create a Virtual Machine

Reviewer: ```suggestion
How to create a virtual machine
```

===============================

> In this examples, we will use the ``vcenter_vm`` module to create a new guest.

Reviewer 2: ```suggestion
In these examples, we will use the ``vcenter_vm`` module to create a new guest.
```

> In this examples, we will use the ``vcenter_vm`` module to create a new guest.

Reviewer: ```suggestion
In this example, we will use the ``vcenter_vm`` module to create a new guest.
```

.. literalinclude:: task_outputs/Create_a_VM.task.yaml

Result
______

.. literalinclude:: task_outputs/Create_a_VM.result.json

.. note::

>     ``vcenter_vm`` accepts much more parameters, however you may prefer to start with a simple VM and use the ``vcenter_vm_hardware`` modules to tune it up afterwards. It's easier this way to identify a potential problematical step.

Reviewer 2: ```suggestion
    ``vcenter_vm`` accepts more parameters than what this example shows.  You may prefer to start with a simple VM and then use the ``vcenter_vm_hardware`` module to configure additional options later. This incremental approach can make it easier to identify a potentially problematical step.
```
I think this section needs to be reworded, I had difficulty understanding it at first.  This suggestion takes some liberties with what you've said here, I'll defer to the docs team on how best to present this information.

In particular, I'm not sure how best to present the idea that just creating a VM with our modules could be problematic and require troubleshooting in an introduction guide.

>     ``vcenter_vm`` accepts much more parameters, however you may prefer to start with a simple VM and use the ``vcenter_vm_hardware`` modules to tune it up afterwards. It's easier this way to identify a potential problematical step.

Reviewer: ```suggestion
    ``vcenter_vm`` accepts more parameters, however you may prefer to start with a simple VM and use the ``vcenter_vm_hardware`` modules to tune it up afterwards. It's easier this way to identify a potential problematical step.
```
